### ISSUE TITLE
Connector cannot be re-used (really sPyNNaker8 but code is here)

### ISSUE THREAD

**Issue by rowleya** [SpiNNakerManchester/sPyNNaker#519]
In PyNN, a connector should be able to be reused in multiple connections.  In PyNN 7, the weight and delay are part of that connection, but in PyNN 8 they are not.  The current implementation sets the weights and delays in the connection when used from PyNN 8, but this means that two connections that use the same connector but different weights and delays will not work; the weights and delays get overwritten.

Example script:
```
p.setup()
inp = p.Population(1, p.SpikeSourceArray(spike_times=[]))
out = p.Population(1, p.IF_curr_exp())

connector = p.OneToOneConnector()

proj_1 = p.Projection(inp, out, connector, p.StaticSynapse(weight=1.0, delay=1.0))
proj_2 = p.Projection(inp, out, connector, p.StaticSynapse(weight=2.0, delay=2.0))

p.run(1)

print proj_1.get(("weight", "delay"), "list")
print proj_2.get(("weight", "delay"), "list")

p.end()
```

**Comment by alan-stokes** [comment]
drat, this is a rather difficult one to solve, till we disconnect from 0.75. 

at moment, i think we'd need to clone the connector (to sticky plaster it), as our low level stuff still assumes weight and delay are in the connector, as they were in 0.75. 

the proper fix would be to remove weights from the connector totally, but thats going to have massive ramifications to the rest of the python, and would require a pynn 0.75 to be either rewritten to accommodate the move, or dropped. 

im guessing we could but a safety check in the connection for the time being, to stop users using it multiple times with different weights / delays.

**Comment by rowleya** [comment]
I think the answer to this really is to remove the weights and delays from the connector, and instead use the PyNN8 concept of having a synapse type which contains these, and which is then passed to the connector at the point it needs them.  This would also involve moving other appropriate methods out of the connector and into the synapse type.  I don't think this is too difficult, though it does involve a fair bit of code mangling.

Note that for PyNN 7, you would then do something like the opposite of what we are currently doing for PyNN 8 i.e. split the given connector into a connector and synapse type.

**Comment by alan-stokes** [comment]
yep. which is why its a rather big change. not something for an afternoon. lol. 
thus why i said maybe in the short term, a safety check can be put in until we get a few days to fix it fully.

**Comment by andrewgait** [comment]
Started looking at this in https://github.com/SpiNNakerManchester/sPyNNaker/pull/523 - have added safety check, will look at fixing it more fully when I have time to do so.

**Comment by alan-stokes** [comment]
merged #523 .so at least we're warning the end users of this now. sorry that my design didnt support this. i honestly didnt know it was a thing.,

**Comment by andrewgait** [comment]
PR #525 has been merged so closing this issue.
